Source/pack.cpp

```diff
@@ -11,10 +11,20 @@
 #include "init.h"
 #include "loadsave.h"
 #include "playerdat.hpp"
+#include "plrmsg.h"
 #include "stores.h"
 #include "utils/endian.hpp"
+#include "utils/log.hpp"
 #include "utils/utf8.hpp"
 
+#define ValidateField(logValue, condition)                                          \

```

Comment:
Consider using the `do while(0)` trick rather than the comma operator:

```c++
#define ValidateField(logValue, condition) \
  do { \
    if (!condition) { \
      LogFailedJoinAttempt(#condition, #logValue, logValue); \
      return false; \
    } \
  } while (0)
```

It's a bit more verbose but perhaps easier to read.